src/client/mod.rs

```diff
@@ -115,6 +115,60 @@ impl APIClient {
         Ok(text)
     }
 
+    pub async fn request_text_stream(
+        &self,
+        request: http::Request<Vec<u8>>,
+    ) -> Result<impl Stream<Item = Result<Vec<u8>>>>
+    {
+        let res: reqwest::Response = self.send(request).await?;
+        trace!("{} {}", res.status().as_str(), res.url());
+
+        // Now use `unfold` to convert the chunked responses into a Stream
+        // We first construct a Stream of Vec<u8> as we potentially might need to
+        // yield multiple objects per loop, then we flatten it to the Stream<Result<u8>> as expected.
+        // Any reqwest errors will terminate this stream early.
```

Comment:
If we are simply streaming items that are chunks of `Vec<u8>`, is buffering necessary?

My initial feeling was that we should either just pass-through the stream entirely and have the client deal with chunks, or we buffer up log lines and emit an event of Item=String per line (but we probably can't do that if we don't know anything about the encoding).

It also feels a bit wrong to divide up huge chunks into stream events into lines when we might get like 500 lines in one chunk. Maybe, that's all okay though. The interface in the example is nice.

I was tempted to just turn each `resp.chunk()` into the stream item because newline buffering would probably interact badly when people try to make "nice" multiline print statements that split the output over multiple lines. But that's probably problems that log aggregators all have to deal with.